src/RandMsg.chpl

```diff
@@ -402,6 +402,38 @@ module RandMsg
         return MsgTuple.error("choice does not support the bigint dtype");
     }
 
+    inline proc logisticGenerator(mu: real, scale: real, ref rs) {
+        var U = rs.next(0, 1);
+
+        while U <= 0.0 {
+            /* Reject U == 0.0 and call again to get next value */
+            U = rs.next(0, 1);
+        }
+        return mu + scale * log(U / (1.0 - U));
+    }
+
+    proc logisticGeneratorMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {
+        const name = msgArgs["name"],
+              isSingleMu = msgArgs["is_single_mu"].toScalar(bool),
+              muStr = msgArgs["mu"].toScalar(string),
+              isSingleScale = msgArgs["is_single_scale"].toScalar(bool),
+              scaleStr = msgArgs["scale"].toScalar(string),
+              size = msgArgs["size"].toScalar(int),
+              hasSeed = msgArgs["has_seed"].toScalar(bool),
+              state = msgArgs["state"].toScalar(int);
+
+        var generatorEntry = st[name]: borrowed GeneratorSymEntry(real);
+        ref rng = generatorEntry.generator;
+        if state != 1 then rng.skipTo(state-1);
+
+        var logisticArr = makeDistArray(size, real);
+        const mu = new scalarOrArray(muStr, !isSingleMu, st),
+              scale = new scalarOrArray(scaleStr, !isSingleScale, st);
+
+        uniformStreamPerElem(logisticArr, rng, GenerationFunction.LogisticGenerator, hasSeed, mu=mu, scale=scale);
+        return st.insert(createSymEntry(logisticArr));
+    }
+
```

Comment:
The code is solid.  I'd love a tutorial on logistic random number generation at one of our aperiodic team meetings.

Reply:
Thanks!! and absolutely! :)